src/features/player-history-aggregator/services/player.history.service.js

```diff
@@ -1,23 +1,80 @@
 import { JSDOM } from "jsdom";
+import { TrackedPlayers } from "../../../models/tracked.players.js";
 import { Players } from "../../../models/players.js";
 
 export function addCurrentPlayersFromSteam(players, games) {
   return games.map((game, i) => {
-    game.playerHistory.push(new Players(players[i]));
+    let existingMonthAndYearIndex = getExistingMonthAndYearIndex(game);
+
+    if (existingMonthAndYearIndex === -1) {
+      game.playerHistory.push(Players.newMonthlyEntry());
+      existingMonthAndYearIndex = game.playerHistory.length - 1;
+    }
+
+    const currentMonthAndYearEntry = game.playerHistory[existingMonthAndYearIndex];
+
+    currentMonthAndYearEntry.trackedPlayers.push(new TrackedPlayers(players[i]));
+
+    currentMonthAndYearEntry.averagePlayers = calculateAveragePlayers(
+      currentMonthAndYearEntry,
+    );
+
     return game;
   });
 }
 
+function getExistingMonthAndYearIndex(game) {
+  const currentYear = new Date().getFullYear();
+  const currentMonth = new Date().getMonth();
+
+  return game.playerHistory.findIndex(
+    (history) => history.year === currentYear && history.month === currentMonth,
+  );
+}
+
+function calculateAveragePlayers(currentMonthAndYearEntry) {
+  const currentTrackedHistories = currentMonthAndYearEntry.trackedPlayers;
+
+  const playersSum = currentTrackedHistories.reduce((previous, current) => {
+    return { players: parseFloat(previous.players) + parseFloat(current.players) };
+  });
+
+  return parseFloat((playersSum.players / currentTrackedHistories.length).toFixed(1));
+}
+
 export function addPlayerHistoriesFromSteamcharts(gamesPagesMap) {
   const games = [];
   for (const [game, page] of gamesPagesMap) {
-    if (page !== "") game.playerHistory = parsePlayerHistory(page);
+    if (page !== "") {
+      const parsedGameHistories = parseGameHistories(page);
+      const fixedGameHistories = Players.manyFromSteamchartsPage(parsedGameHistories);
```

Comment:
Create new method on the games class -> `games.addManyHistoryEntries(parsedGameHistories)`.

In this method, no need for `forEach`, you can use: `game.playerHistory.push(...Players.manyFrom...);` and then do the sorting in the same method or a private method.